src/mds/CInode.cc

```diff
@@ -4545,8 +4552,13 @@ void CInode::validate_disk_state(CInode::validated_data *results,
 					      &equivalent, &divergent);
 
       if (divergent || memory_newer < 0) {
-	// we're divergent, or on-disk version is newer
-	results->backtrace.error_str << "On-disk backtrace is divergent or newer";
```

Comment:
'memory_newer < 0' is error, shouldn't be ignored

Reply:
@ukernel 
I've just added the check inside the `if (divergent || memory_newer < 0)` condition to test specifically for `if (divergent && in->is_dirty_parent())`.
The else part sets the error string just as it would if `memory_newer < 0` in the older condition check. The `memory_newer < 0` condition is not ignored.